tools/hlt_client/hlt_client/download_game.py

```diff
@@ -128,30 +128,41 @@ class UserGameDownloader(GameDownloader):
     _FETCH_THRESHOLD = 250
     _BUCKETS = []
 
-    def __init__(self, destination, user_id, limit):
+    def __init__(self, destination, user_id, limit, date):
         """
         Download games for a user
         :param destination: Where to download
         :param user_id: Which user's replays to fetch
         :param limit: How many replays to fetch (max)
+        :param date: Which date to download
         """
         self.destination = destination
-        self.objects = self._parse_user_metadata(self._fetch_metadata(user_id, limit))
+        self.objects = self._parse_user_metadata(self._fetch_metadata(user_id, limit, date))
 
-    def _fetch_metadata(self, user_id, limit):
+    def _fetch_metadata(self, user_id, limit, date):
         """
         Retrieves paginated game metadata from the halite servers for a specified user up to limit items
         :param user_id: The id of the user to fetch
         :param limit: The maximum number of items to fetch
+        :param date: Which date to download
         :return: The full metadata of items
         """
         print('Fetching Metadata')
         current = 0
         result_set = []
         while current <= limit:
-            current_limit = self._FETCH_THRESHOLD if ((limit - current) >= self._FETCH_THRESHOLD) else (limit - current)
-            result_set += requests.get(self._USER_BOT_URI.format(user_id, current_limit, current)).json()
-            current += self._FETCH_THRESHOLD
+            if date is None:
+                current_limit = self._FETCH_THRESHOLD if ((limit - current) >= self._FETCH_THRESHOLD) else (limit - current)
+                result_set += requests.get(self._USER_BOT_URI.format(user_id, current_limit, current)).json()
+                current += self._FETCH_THRESHOLD
+            else:
+                if requests.get(self._USER_BOT_URI.format(user_id, 1, current)).json()[0]["replay"].split("-")[1] != date:
```

Comment:
Magic number

Reply:
I was trying to check the date of the retrieved replay one at a time. Do you have any suggestion on how to do this faster? I can also retrieve current_limit replays at a time and filter out replays that belongs to the wrong date and append the filtered list to the list result_set.